nemo/src/model/types/primitive_types.rs

```diff
@@ -130,6 +130,30 @@ impl TryFrom<NestedType> for PrimitiveType {
 }
 
 impl PrimitiveType {
+    // TODO: I think this should be the PartialCmp between types but as long as we do not
+    // have casting we still want to forbid e.g. integer and string merges while reasoning
+    // HOWEVER for data that occurs in facts or sources we can decide for the max type up
+    // front and read them accordingly
+    /// Get the more general type out of two types (not necessarily castable but can be used to
+    /// determine how data should be read)
+    pub fn max_type(&self, other: &PrimitiveType) -> PrimitiveType {
+        match self {
+            PrimitiveType::Any => PrimitiveType::Any,
+            PrimitiveType::String => match other {
+                PrimitiveType::String => PrimitiveType::String,
+                _ => PrimitiveType::Any,
+            },
+            PrimitiveType::Integer => match other {
+                PrimitiveType::Integer => PrimitiveType::Integer,
+                _ => PrimitiveType::Any,
```

Comment:
I think we should handle the case where we have `Integer` and `Decimal` here (which should promote to `Decimal`, _not_ to `Any`).

Reply:
There is no `Decimal` variant in `PrimitiveType`.

Reply:
Ah, right.